drivers/hmis_csv_twenty_twenty_two/app/models/hmis_csv_twenty_twenty_two/exporter/base.rb

```diff
@@ -0,0 +1,667 @@
+      @file_path = "#{file_path}/#{Time.now.to_f}"
+      @logger = logger
+      @debug = debug
+      @range = ::Filters::DateRange.new(start: start_date, end: end_date)
+      @projects = projects
+      @period_type = period_type.presence || 3
+      @directive = directive.presence || 2
+      @hash_status = hash_status.presence || 1
+      @faked_pii = faked_pii
+      @user = ::User.find(user_id)
+      @include_deleted = include_deleted
+      @faked_environment = faked_environment
+    end
+
+    def export!
+      create_export_directory
+      begin
+        set_time_format
+        setup_export
+
+        # Project related items
+        export_projects
+        export_project_cocs
+        export_organizations
+        export_inventories
+        export_funders
+        export_affiliations
+
+        # Enrollment related
+        export_enrollments
+        export_exits
+        export_clients
+        export_enrollment_cocs
+        export_disabilities
+        export_employment_educations
+        export_health_and_dvs
+        export_income_benefits
+        export_services
+        export_current_living_situations
+        export_assessments
+        export_assessment_questions
+        export_assessment_results
+        export_events
+        export_youth_education_statuses
+
+        export_users
+
+        build_export_file
+        zip_archive
+        upload_zip
+        save_fake_data
+      ensure
+        remove_export_files
+        reset_time_format
+      end
+      @export
+    end
+
+    def set_time_format
+      # We need this for exporting to the appropriate format
+      @default_date_format = Date::DATE_FORMATS[:default]
+      @default_time_format = Time::DATE_FORMATS[:default]
+      Date::DATE_FORMATS[:default] = '%Y-%m-%d'
+      Time::DATE_FORMATS[:default] = '%Y-%m-%d %H:%M:%S'
+    end
+
+    def reset_time_format
+      Date::DATE_FORMATS[:default] = @default_date_format
+      Time::DATE_FORMATS[:default] = @default_time_format
+    end
+
+    def save_fake_data
+      return unless @faked_pii
+
+      @export.fake_data.save
+    end
+
+    def zip_path
+      @zip_path ||= File.join(@file_path, "#{@export.export_id}.zip")
+    end
+
+    def csv_file_path(klass)
+      File.join(@file_path, klass.hud_csv_file_name(version: version))
+    end
+
+    def upload_zip
+      @export.file = Pathname.new(zip_path).open
+      @export.content_type = @export.file.content_type
+      @export.content = @export.file.read
+      @export.save
+    end
+
+    def zip_archive
+      files = Dir.glob(File.join(@file_path, '*')).map { |f| File.basename(f) }
+      Zip::File.open(zip_path, Zip::File::CREATE) do |zipfile|
+        files.each do |filename|
+          zipfile.add(
+            File.join(@export.export_id, filename),
+            File.join(@file_path, filename),
+          )
+        end
+      end
+    end
+
+    def remove_export_files
+      FileUtils.rmtree(@file_path) if File.exist? @file_path
+    end
```

Comment:
I wonder if it makes sense to move some of this to a concern, even if we don't back-port it to the 2020 version, it would mean much less code review for these in the future, unless they change.